services/api/src/models/user.js

```diff
@@ -16,6 +16,22 @@ schema.virtual('password').set(function (password) {
   this._password = password;
 });
 
+schema.methods.addAuthToken = function ({ exp, jti, iat }, ctx) {
+  // filter out any expired tokens + prevent that we are not adding a duplicate
+  const authTokens = (this.authTokens || []).filter((token) => token.jti !== jti && token.exp > Date.now());
+
+  this.authTokens = [
+    {
+      exp: new Date(exp * 1000),
```

Comment:
I'm assuming that the need for this is to be able to use mongo to query on the expired date? is that right? I find it a bit odd that we're passing in and storing the raw payload instead of the token itself but this is the reason right? (in other words we need to be able to query on the payload data itself)

Reply:
Re the date not planning on query it via mongo but sure thats an option, just easier to have it store as date, dont need to convert from jwt date to actually date other places. 

```
I find it a bit odd that we're passing in and storing the raw payload instead of the token itself but this is the reason right? (in other words we need to be able to query on the payload data itself)
```

I don't need the actually token, would just be another thing that could leak. There is nothing I am storing in `authTokens` that cant be exposed.
 
i want to allow the user to identify a token (ip, useragent) and being able to revoke it (jti).
And then the exp is used to avoid the list growing too much.

Reply:
well it just seems slightly odd to be storing each part of the payload separately instead of an array of tokens as strings as these already are packaged to contain the payload (that's what they're designed for). The one exception here I can think of is if you needed to query on separated parts of the payload...

Reply:
ok nvm after discussion this was my misunderstanding